src/appMain/life_cycle.cc

```diff
@@ -119,7 +120,9 @@ bool LifeCycle::StartComponents() {
   }
 
 #ifdef ENABLE_SECURITY
-  security_manager_ = new security_manager::SecurityManagerImpl();
+  utils::SystemTimeHandler* handler =
+      new application_manager::SystemTimeHandlerImpl(*app_manager_);
+  security_manager_ = new security_manager::SecurityManagerImpl(handler);
```

Comment:
@AByzhynar would be nice to ad comment that security_manager_ is responsible for SystemTimeHandler lifetime